### DISCUSSION TITLE
How to use CoolProp as default

### DISCUSSION THREAD

**Discussion by ssosas** [petrobras/ccp#16] [Q&A]
Hi all,

I've installed CCP but I do not plan to get REFPROP library. Is there any way to configure CCP to call exclusively CoolProp? In my case I will only use AIR as fluid. I have tried to modify ccp/config/__init__.py (EOS="HEOS") but I still getting errors/warnings about REFPROP missing :(

Many thanks!

Sergio

**Comment by raphaeltimbo** [comment]
Hi @ssosas !
Could you share what errors are you getting? In theory, it should be possible to run basic cases without REFPROP. As an example, the code that is available in the README I can run like this (notice the second line where I set the EOS):

```python
import ccp

ccp.config.EOS = "HEOS"  # no need to modify the init file directly, you can set the eos here.

# ccp uses pint to handle units. Q_ is a pint quantity.
# If a pint quantity is not provided, SI units are assumed.
Q_ = ccp.Q_
ps = Q_(3, 'bar')
Ts = 300

# Define the fluid as a dictionary:
fluid = {
    "CarbonDioxide": 0.8,
    "Nitrogen": 0.2,
}

# Define suction and discharge states:

suc0 = ccp.State(fluid=fluid, p=ps, T=Ts)
disch0 = ccp.State(fluid=fluid, p=Q_(7.255, 'bar'), T=391.1)
disch1 = ccp.State(fluid=fluid, p=Q_(6.754, 'bar'), T=382.1)

# Create performance point(s):

point0 = ccp.Point(
    suc=suc0,
    disch=disch0,
    speed=Q_(7941, 'RPM'),
    flow_m=Q_(34203.6, 'kg/hr'),
    b=0.0285,
    D=0.365,
)
point1 = ccp.Point(
    suc=suc0,
    disch=disch1,
    speed=Q_(7941, 'RPM'),
    flow_m=Q_(36204.8, 'kg/hr'),
    b=0.0285,
    D=0.365,
)

# Create an impeller with those points:

imp = ccp.Impeller([point0, point1])

# Get results from the Impeller with methods such as
imp.head_plot()
imp.disch.T_plot()

```

After I import ccp I still get the warning, but the code will run just fine.

As I said, most basic things should run without REFPROP, but I have to be honest that I did not test everything since I am always using REFPROP.

In the future we could try to work on some of the issues that we have when we use only CoolProp if there is a demand for that (we would also accept contributions on that).

**Reply by ssosas** [reply]
Many thanks @raphaeltimbo for this quick answer. I use the code in MacBook, the code exists with 0 so I think there is no error/exception but just the warning about REFPROP (that is printed with the keyword ERROR :) on the pycharm terminal), but at the end I got no plots so I don't know, I need to do more tests.... In the past I also had some problems on my Mac when plotting with python so let me try CCP in my Debian laptop and I will comeback to you!! 
Cheers!
Sergio

**Reply by raphaeltimbo** [reply]
Hi @ssosas !
We use plotly to create the plots which are html based.
To show them inside pycharm, you need to run the code inside a notebook:
![image](https://github.com/petrobras/ccp/assets/18506378/5a7fbec5-68a0-4cb1-b542-a2569db90f73)

Or you could modify the code to save the plots to html files:

```python
head_fig = imp.head_plot()
disch_T_fig = imp.disch.T_plot()

head_fig.write_html("head.html")
disch_T_fig.write_html("disch_T.html")
```

**Reply by ssosas** [reply]
Hi again Raphael, so with your last indications it works fine ;) I can generate html files containing plots :) Many thanks. I have also tested with jupyter but no luck :( plots are not shown, just a white space... very strange. I will keep you informed!

**Reply by ssosas** [reply]
Hi @raphaeltimbo! Hope you are doing well ;)

Could you please tell me which diameter we need to set when defining the impeller? Is it the outlet diameter? The inlet diameter? The mean between inlet/outlet? 

Thanks a lot ;)

**Reply by raphaeltimbo** [reply]
Hi @ssosas! It is the outlet diameter.
I will update the documentation to mention that.

**Reply by Hempstead** [reply]
Hi, all, I have installed ccp and trying to the tutorial. I got the following error message:

 ---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
Cell In[6], line 50
     48 # Get results from the Impeller with methods such as
     49 imp.head_plot()
---> 50 imp.disch.T_plot()

File ~\.conda\envs\comp\lib\site-packages\ccp\config\units.py:187, in check_units.<locals>.inner(*args, **kwargs)
    184     else:
    185         base_unit_kwargs[k] = v
--> 187 return func(*base_unit_args, **base_unit_kwargs)

File ~\.conda\envs\comp\lib\site-packages\ccp\impeller.py:121, in ImpellerPlotFunction.__call__(self, flow_v, speed, plot_kws, *args, **kwargs)
    118 surge_flow_list = []
    119 surge_attr_list = []
    120 attr_units = kwargs.get(
--> 121     f"{attr}_units", getattr(impeller_object.curves[0], attr).units
    122 )
    124 for curve in impeller_object.curves:
    125     fig = r_getattr(curve, attr + "_plot")(fig=fig, plot_kws=plot_kws, **kwargs)

AttributeError: 'Curve' object has no attribute 'disch.T'

Can you help?
